core/thread/inc/ROOT/TThreadedObject.hxx

```diff
@@ -100,34 +95,26 @@ namespace ROOT {
             }
          };
 
-         template<class T, bool ISHISTO = std::is_base_of<TH1,T>::value>
-         struct DirCreator{
-            static std::vector<TDirectory*> Create(unsigned maxSlots) {
-               std::string dirName = "__TThreaded_dir_";
-               dirName += std::to_string(ROOT::Internal::TThreadedObjectUtils::GetTThreadedObjectIndex()) + "_";
-               std::vector<TDirectory*> dirs;
-               dirs.reserve(maxSlots);
-               for (unsigned i=0; i< maxSlots;++i) {
-                  auto dir = gROOT->mkdir((dirName+std::to_string(i)).c_str());
-                  dirs.emplace_back(dir);
-               }
-               return dirs;
+         template <class T, bool ISHISTO = std::is_base_of<TH1, T>::value>
+         struct DirCreator {
+            static TDirectory *Create()
+            {
+               static unsigned dirCounter = 0;
+               const std::string dirName = "__TThreaded_dir_" + std::to_string(dirCounter++) + "_";
+               return gROOT->mkdir(dirName.c_str());
             }
          };
 
-         template<class T>
-         struct DirCreator<T, true>{
-            static std::vector<TDirectory*> Create(unsigned maxSlots) {
-               std::vector<TDirectory*> dirs(maxSlots, nullptr);
-               return dirs;
-            }
+         template <class T>
+         struct DirCreator<T, true> {
+            static TDirectory *Create() { return nullptr; }
          };
 
          struct MaxSlots_t {
             unsigned fVal;
             operator unsigned() const { return fVal; }
             MaxSlots_t &operator=(unsigned val)
-               R__DEPRECATED(6, 24, "Superior interface exists: construct TThreadedObject using TNumSlots instead.")
```

Comment:
`R__DEPRECATED` is the user API :-)

Reply:
will fix (here and elsewhere, also my fault :D)